centreon/src/Core/Common/Infrastructure/Repository/ValuesByPackets.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types = 1);
+
+namespace Core\Common\Infrastructure\Repository;
+
+/**
+ * This class is used to split a 'GET' API call into multiple calls, due to URL size limitations.
+ * The main cause of this division is the size of parameter lengths.
```

Comment:
I think I don't understand well what it really does.
I don't know to explain, so here a brief summary : my mind is still in "why ?" posture.

Reply:
Because there's a limitation on uri size when calling Cloud APIs.
It's a Cloud restriction.

Reply:
I am not "why ?" about the URI limitation, it exists everwhere outside (ex: apache has a default limit around 8KB).

But I don't see how we have a equivalence between
- 1 api call with a long uri with bunch of GET parameters
- N api calls because we splited the GET parameters in batchs
:point_up: That's what I understand about the phpdoc etc ...